src/wic/api/pythonapi.py

```diff
@@ -335,24 +368,30 @@ def yaml(self) -> dict[str, Any]:
         return d
 
     def _save_yaml(self) -> None:
-        Path(self.path).mkdir(parents=True, exist_ok=True)
-        self.yml_path = self.path.joinpath(f"{self.name}.yml")
+        _WIC_PATH.mkdir(parents=True, exist_ok=True)
+        self.yml_path = _WIC_PATH.joinpath(f"{self.name}.yml")
         with open(self.yml_path, "w", encoding="utf-8") as file:
             file.write(yaml.dump(self.yaml))
 
     def _save_all_cwl(self) -> None:
-        Path(self.path).mkdir(parents=True, exist_ok=True)
+        """Save CWL files to cwl_adapters.
+
+        This is necessary for WIC to compile the workflow.
+        """
+        _WIC_PATH.mkdir(parents=True, exist_ok=True)
         for s in self.steps:
             try:
-                s._save_cwl(self.path)  # pylint: disable=W0212
+                s._save_cwl(_WIC_PATH)  # pylint: disable=W0212
             except BaseException as exc:
                 raise exc
 
     def compile(self, run_local: bool = False) -> Path:
-        """Compile Workflow using WIC."""
-        # The CommandLineTools should already be on disk, and moreover when
-        # self.path = '.' this will overwrite them!
-        # self._save_all_cwl()
+        """Compile Workflow using WIC.
+
+        Returns path to compiled CWL Workflow.
+        """
+
+        self._save_all_cwl()
```

Comment:
This line still erroneously overwrites CWL files if they are already on disk. (See comment above which you have deleted.)

This is a problem if the new CWL file contents are loaded from a remote URI (and/or otherwise incorrectly autogenerated) and then the CWL file which has been hand-crafted by an expert (i.e. me) to be correct gets overwritten.